### PR TITLE
Display history info

### PR DESCRIPTION
**Pull request by vittoriaThinkst**
## Proposed changes

Map alert history objects
Handle exceptions
Add tests

Screenshot 2024-05-03 at 10 33 36
Screenshot 2024-05-03 at 10 33 45
Screenshot 2024-05-03 at 10 34 10

### CODE REVIEW

File: frontend_vue/src/components/constants.ts [hunk 1]
```diff
@@ -66,3 +67,21 @@ export const INCIDENT_LIST_EXPORT = {
   JSON: 'incidentlist_json',
   CSV: 'incidentlist_csv',
 };
+
+export const INCIDENT_CHANNEL_TYPE_LABELS = {
+  aws_keys: 'AWS API Key Token',
+  cc: 'Credit Card Token',
+  azure_id: 'Azure Login Certificate Token',
+};
```
Comment (mclmax) [inline frontend_vue/src/components/constants.ts:75]: this seems wrong to me. these aren't input channels. the input channels are:

```
INPUT_CHANNEL_HTTP = "HTTP"
INPUT_CHANNEL_DNS = "DNS"
INPUT_CHANNEL_IMGUR = "Imgur"
INPUT_CHANNEL_LINKEDIN = "LinkedIn"
INPUT_CHANNEL_BITCOIN = "Bitcoin"
INPUT_CHANNEL_SMTP = "SMTP"
INPUT_CHANNEL_MTLS = "Kubeconfig"
INPUT_CHANNEL_MYSQL = "MYSQL"
INPUT_CHANNEL_WIREGUARD = "WireGuard"
```

These seem more like token type labels?

File: frontend_vue/src/utils/incidentAlertService.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        } else {
+          acc[key] = value;
+        }
+      }
+      return acc;
+    }, {} as FormattedHitsType);
+  } catch (error) {
+    console.error(`Error in remove Null, Empty Objects and Arrays: ${error}`);
+    return incidentDetails;
+  }
```
Comment (mclmax) [inline frontend_vue/src/utils/incidentAlertService.ts:109]: i'm still unsure why we need to do this? can these checks not just be in the html render steps? v-`if="!notEmpty(key) && !notEmpty(value)"` kinda thing?

File: frontend_vue/src/utils/incidentAlertService.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+function isCreditCardtoken(token: string) {
+  return token === TOKENS_TYPE.CREDIT_CARD;
+}
+
+function isAWStoken(token: string) {
+  return token === TOKENS_TYPE.AWS_KEYS;
+}
+
+export function buildIncidentDetails(
```
Comment (mclmax) [inline frontend_vue/src/utils/incidentAlertService.ts:120]: same for this function, what do we get from doing this here instead of in the rendering step? most of this is just `'thing': hitAlert.thing`?

**Review by mclmax — CHANGES_REQUESTED** [review]

File: frontend_vue/src/components/constants.ts (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/constants.ts:75]: That's what I found in the current forntend setup:
`  {% if canarydrop['type'] == 'aws_keys' %}
                                  <b>Channel:</b> AWS API Key Token
  {% elif canarydrop['type'] == 'cc' %}
                                  <b>Channel:</b> Credit Card Token
  {% elif canarydrop['type'] == 'azure_id' %}
                                  <b>Channel:</b> Azure Login Certificate Token
  {% else %}
                                  <b>Channel:</b> {{ canarydrop['triggered_list'][item]['input_channel'] }}
  {% endif %}`
  
  It looks like we want to display a custom channel label for these specific token types.
  If these variables exist on the backend, we can maybe implement them for these token types as well. Instead of adding this channel check on the frontend side

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 2 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/utils/incidentAlertService.ts:109]: yeah for sure I can do it in the template. I was just happier by dividing concerns if possible. I would like to avoid cluttering the template since we are already dealing with 3 levels of object nesting. I'm afraid adding even another condition would make it quite messy to read/understand.

Comment (vittoriaThinkst) [reply frontend_vue/src/utils/incidentAlertService.ts:109]: That said, if you think having the check on the template makes sense, I will move to that solution

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 3 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/utils/incidentAlertService.ts:120]: most of them, but not all of them.
We have many conditional fields here, and I'm happy to deal with them before I pass the object to the UI.
I'd have to check those conditions on the template side otherwise. This would add a lot of complexity and would make the template cluttered and hard to follow

ie.
`geo_info: !hitAlert.geo_info.bogon
        ? {
            ...hitAlert.geo_info,
          }
        : {
            ip: hitAlert.geo_info.ip,
            bogon: hitAlert.geo_info.bogon,
          },`

File: frontend_vue/src/components/constants.ts (hunk 1 above)
Comment (mclmax) [reply frontend_vue/src/components/constants.ts:75]: what do the other tokens display for Channel on old tokens.org?

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 2 above)
Comment (mclmax) [reply frontend_vue/src/utils/incidentAlertService.ts:109]: we can try it out. it is a little difficult to read. but maybe we don't need to worry about it

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 3 above)
Comment (mclmax) [reply frontend_vue/src/utils/incidentAlertService.ts:120]: we can try it. i in my head i figured it would be clearer in the html

File: frontend_vue/src/components/constants.ts (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/constants.ts:75]: heh, that would be good to know. I'm not sure how to test most of the tokens, all I can rely on is the current history file

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 3 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/utils/incidentAlertService.ts:107]: I'm really not sure how to handle this inside the template.
It means I'll have to add checks at many points in the nesting loops, checking for specific key/vals

for instance for the bogon, I will have to add something like 
`                <ul
                  v-if="isNotEmpty(nested_val) && hitAlert.geo_info.bogon"
                  class="break-words"
                >
                  <IncidentDetailsListItem
                    :label="nested_key"
                    :value="hitAlert.geo_info.bogon"
                  />
                  <IncidentDetailsListItem
                    :label="nested_key"
                    :value="hitAlert.geo_info.ip,"
                  />
                  </ul>
               <template v-else>
               <IncidentDetailsListItem
                    v-if="!isObject(nested_val) "
                    :label="nested_key"
                    :value="nested_val"
                  />
                  </template>
                  ... move forward
                  
 We also add an extra 'basic info' key, which arbitrarily gathers some values that the backend gives us not ordered
       basic_info: {
        token_type: formatTokenTypeLabel(hitAlert.token_type),
        input_channel: hasChannelCustomLabel(hitAlert.input_channel),
        src_data: hitAlert.src_data,
        useragent: hitAlert.useragent || null,
        etc
      },`
 `

Comment (vittoriaThinkst) [reply frontend_vue/src/utils/incidentAlertService.ts:107]: There's no way I can think of, that this would look fine on a template

File: frontend_vue/src/components/constants.ts (hunk 1 above)
Comment (mclmax) [reply frontend_vue/src/components/constants.ts:75]: So here's an example. Is this the correct place we're using it? How are we handling the DNS channel? Looking at your code it looks like we aren't?

Screenshot 2024-05-08 at 13 51 11

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 3 above)
Comment (mclmax) [reply frontend_vue/src/utils/incidentAlertService.ts:107]: let's skip it. by "we can try it" i meant lets go with your code heh

File: frontend_vue/src/utils/incidentAlertService.ts (hunk 2 above)
Comment (mclmax) [reply frontend_vue/src/utils/incidentAlertService.ts]: heh i meant let's try your way but i see you've moved it to the template?

**Review by mclmax — APPROVED** [review]
